src/viam/app/data_client.py

```diff
@@ -659,25 +659,27 @@ async def streaming_data_capture_upload(
         )
         metadata = DataCaptureUploadMetadata(upload_metadata=upload_metadata, sensor_metadata=sensor_metadata)
         request_metadata = StreamingDataCaptureUploadRequest(metadata=metadata)
+        stream: Stream[StreamingDataCaptureUploadRequest, StreamingDataCaptureUploadResponse]
         async with self._data_sync_client.StreamingDataCaptureUpload.open(metadata=self._metadata) as stream:
             await stream.send_message(request_metadata)
             await stream.send_message(StreamingDataCaptureUploadRequest(data=data), end=True)
-            response: StreamingDataCaptureUploadResponse = await stream.recv_message()
+            response = await stream.recv_message()
             if not response:
                 await stream.recv_trailing_metadata()  # causes us to throw appropriate gRPC error
+                raise TypeError("Response cannot be empty")
```

Comment:
(q) Is it possible for us to actually reach this point? `if not response` is the case when the response isn't just empty, but a `None`, which I think should only happen if something went wrong such that we'd have a non-OK gRPC status and so throw an error at the preceding line. I guess either way there's no harm in having another safety rail here!

Reply:
*WE* know we can't get here, but they *typechecker* doesn't know that and it was complaining

Reply:
Ahh got it, thanks!